docs/user_guide.rst

```diff
@@ -131,6 +131,39 @@ main seed mechanism available:
   To override the filesystem location of the seed cache, one can use the
   ``VIRTUALENV_OVERRIDE_APP_DATA`` environment variable.
 
+Bundled wheels
+~~~~~~~~~~~~~~
+
+What qualifies as bundled wheel? The following three sets together:
+
+- ``virtualenv`` ships out of box with a set of ``wheels`` for all three seed packages (:pypi:`pip`, :pypi:`setuptools`,
+  :pypi:`wheel`). We refer to these as embedded wheels. Different Python versions require different versions of these,
+  and because virtualenv supports a wide range of Python versions, the number of embedded wheels out of box is greater
+  than 3.
+- ``virtualenv`` updates the embedded wheels fairly quickly after the upstream packages release a new version, however
+  end users might not be able to upgrade virtualenv at the same speed. Therefore, a user might request to upgrade the
+  list of embedded wheels by invoking virtualenv with the :option:`upgrade-embed-wheels` flag. This operation will
+  trigger automatically, as a background process, if no upgrade has been performed in the last 14 days and upgrade the
+  embedded wheels if they have been our for more than 28 days. This 28 days period should guarantee end users are not
+  pulling in automatically releases that have known bugs within. This automatic behaviour might be disabled via the
+  :option:`no-periodic-update` configuration flag/option. To acquire the release date of a package virtualenv will
+  perform the following:
+
+  - lookup ``https://pypi.org/pypi/<distribution>/json`` (primary truth source),
+  - save the date the version was first discovered, and wait until 28 days passed.
```

Comment:
On the JSON page, `data["releases"][version]["upload_time_iso_8601"]` should get us this information. :)

The question is whether 28 days since release is better, or 28 days since release seen is better? (also, how careful do we want to be, that we don't block upgrades if there's a release every 27 days or something like that)

Reply:
If you take a look at the implementation code, see https://github.com/gaborbernat/virtualenv/blob/bundle/src/virtualenv/seed/wheels/periodic_update.py, this is already implemented. We take 28 days since release if available, otherwise 28 days since seen. This ensures at least 28 days of grace period which is what we want here.

> also, how careful do we want to be, that we don't block upgrades if there's a release every 27 days or something like that

This is not a problem because in such cases we'll pick the first available minor that's at least 28 days old, see https://github.com/gaborbernat/virtualenv/blob/bundle/src/virtualenv/seed/wheels/periodic_update.py#L44